### ISSUE TITLE
systemd Files have wrong Owner / Group

### ISSUE THREAD

**Issue by erisd** [bigbluebutton/bigbluebutton#10831]
**Describe the bug**
systemd Files have wrong Owner / Group - like etherpad or red5

**To Reproduce**
```
ls -la /usr/lib/systemd/system/etherpad.service
-rw-r--r-- 1 etherpad etherpad 448 Sep 26 13:05 /usr/lib/systemd/system/etherpad.service
ls -la /usr/lib/systemd/system/red5.service
-rw-r--r-- 1 red5 red5 287 Dec 10  2019 /usr/lib/systemd/system/red5.service
```

**Expected behavior**
this files MUST be owned by root.

**Comment by GhaziTriki** [comment]
Dear @erisd,
I don't think it an issue, in the contrary they should be owned by root and the services should not be run by root user. Can you please give more arguments why they should be owned by root?

**Comment by erisd** [comment]
I think I  lost you...
Where is the link between the owner / group of the systemd Unit and the user under which systemd will start the Service?
With the actual state a application which runs as meteor can modify its service unit (like remove user / group) and kill it self. Systemd will then restart the process, but now as root. (Its very useful for a exploit chain, but I hope thats not your target)

**Comment by GhaziTriki** [comment]
Sorry, in my comment I meant they should **NOT** be owned by root.

**Comment by danimo** [comment]
@GhaziTriki I think there might be an important misunderstand here:

1. The ownership of the system service file signifies who can edit/alter it. Having any other user than root makes it a perfect part of an exploit chain (assuming the service gets compromised, a root shell is only one service restart away). This is what @erisd is referring to.
2. [`User=` and `Group=`](https://www.freedesktop.org/software/systemd/man/systemd.exec.html#User=) within a `.service` file: This is what actually specifies the effective user and group that the service will be run as.

We need to fix 1. to prevent exploit chains. There is more tweaking that can be done in the service file to isolate the running process from the system, but first things first.

**Comment by ffdixon** [comment]
Fixed in upcoming 2.2.31.

# ls -la /usr/lib/systemd/system/etherpad.service
-rw-r--r-- 1 root root 448 Dec  1 01:56 /usr/lib/systemd/system/etherpad.service

# ls -la /usr/lib/systemd/system/red5.service
-rw-r--r-- 1 root root 287 Dec 12  2018 /usr/lib/systemd/system/red5.service

**Comment by erisd** [comment]
Thanks! But all systemd units, not only this examples - correct?
```
ls -la /usr/lib/systemd/system/
total 68
drwxr-xr-x 2 red5     red5     4096 Dec  1 09:53 .
drwxr-xr-x 8 root     root     4096 May  6  2020 ..
-rw-r--r-- 1 meteor   meteor    458 Nov 27 18:20 bbb-html5.service
-rw-r--r-- 1 root     root      342 Sep 25 18:56 bbb-rap-archive-worker.service
-rw-r--r-- 1 root     root      224 Sep 25 18:56 bbb-rap-caption-inbox.service
-rwxr-xr-x 1 root     root      337 Sep 25 18:56 bbb-rap-events-worker.service
-rw-r--r-- 1 root     root      340 Sep 25 18:56 bbb-rap-process-worker.service
-rw-r--r-- 1 root     root      343 Sep 25 18:56 bbb-rap-publish-worker.service
-rw-r--r-- 1 root     root      346 Sep 25 18:56 bbb-rap-sanity-worker.service
-rw-r--r-- 1 root     root       56 Sep 25 18:56 bbb_record_core.slice
-rwxr-xr-x 1 root     root      246 Sep 25 18:56 bbb-record-core.target
-rw-r--r-- 1 root     root      175 Sep 25 18:56 bbb-record-core.timer
-rw-r--r-- 1 root     root      412 Nov 19 21:45 bbb-webrtc-sfu.service
-rw-r--r-- 1 meteor   meteor    293 Nov 27 18:20 disable-transparent-huge-pages.service
-rw-r--r-- 1 etherpad etherpad  448 Sep 26 15:05 etherpad.service
-rw-r--r-- 1 root     root      503 Nov 19 21:45 kurento-media-server.service
-rw-r--r-- 1 red5     red5      287 Dec 10  2019 red5.service
```
the directory permission should also not touched by red5.

If youre already on this:
Can you change also the location of bbb-web.service and freeswitch.service? Actually they are in /lib/systemd/system/ - but that location is for the distributor. That would be great, then the bbb related systemd units are in one folder...
https://unix.stackexchange.com/questions/206315/whats-the-difference-between-usr-lib-systemd-system-and-etc-systemd-system

**Comment by ffdixon** [comment]
Fixed the directory permissions in the upcoming 2.2.31 version.

~~~
# ls -la /usr/lib/systemd/system/
total 80
drwxr-xr-x  2 root root 4096 Dec  6 22:28 .
drwxr-xr-x 10 root root 4096 Dec  6 22:25 ..
-rw-r--r--  1 root root  458 Dec  6 22:11 bbb-html5.service
-rw-r--r--  1 root root  342 Dec  6 21:44 bbb-rap-archive-worker.service
-rw-r--r--  1 root root  224 Dec  6 21:44 bbb-rap-caption-inbox.service
-rw-r--r--  1 root root  337 Dec  6 21:44 bbb-rap-events-worker.service
-rw-r--r--  1 root root  340 Dec  6 21:44 bbb-rap-process-worker.service
-rw-r--r--  1 root root  343 Dec  6 21:44 bbb-rap-publish-worker.service
-rw-r--r--  1 root root  346 Dec  6 21:44 bbb-rap-sanity-worker.service
-rw-r--r--  1 root root   56 Dec  6 21:44 bbb_record_core.slice
-rw-r--r--  1 root root  246 Dec  6 21:44 bbb-record-core.target
-rw-r--r--  1 root root  175 Dec  6 21:44 bbb-record-core.timer
-rw-r--r--  1 root root  412 Nov 19 20:45 bbb-webrtc-sfu.service
-rw-r--r--  1 root root  293 Dec  6 22:11 disable-transparent-huge-pages.service
-rw-r--r--  1 root root  448 Dec  5 02:05 etherpad.service
-rw-r--r--  1 root root  725 Dec  6 22:29 kurento-media-server-8888.service
-rw-r--r--  1 root root  725 Dec  6 22:29 kurento-media-server-8889.service
-rw-r--r--  1 root root  725 Dec  6 22:29 kurento-media-server-8890.service
-rw-r--r--  1 root root  503 Nov 19 20:45 kurento-media-server.service
-rw-r--r--  1 root root  287 Dec 10  2019 red5.service
~~~


> Can you change also the location of bbb-web.service and freeswitch.service? Actually they are in /lib/systemd/system/ - but that location is for the distributor. That would be great, then the bbb related systemd units are in one folder...

Need to look at doing this in 2.3.

**Comment by paulmenzel** [comment]
As commented in issue https://github.com/bigbluebutton/bigbluebutton/issues/11752, there are two service files left in BBB 2.3.3.

**Comment by antobinary** [comment]
Retagging to 2.5 where the build/ directory contains all the packaging related files (open source). Once resolved we can backport to all maintained versions

**Comment by schrd** [comment]
I think this can be closed as the packages have the correct permissions on unit files on a 2.4 system with packages from the public packaging scripts:
~~~
ls -la /usr/lib/systemd/system/
total 52
drwxr-xr-x  2 root root 4096 Jan  3 14:19 .
drwxr-xr-x 12 root root 4096 Sep  3 11:13 ..
-rw-r--r--  1 root root  564 Jan  3 14:14 bbb-html5-backend@.service
-rw-r--r--  1 root root  553 Jan  3 14:14 bbb-html5-frontend@.service
-rw-r--r--  1 root root  558 Jan  3 14:14 bbb-html5.service
-rw-r--r--  1 root root  262 Dec 23 12:24 bbb-rap-caption-inbox.service
-rw-r--r--  1 root root  511 Dec 23 12:24 bbb-rap-resque-worker.service
-rw-r--r--  1 root root  307 Dec 23 12:24 bbb-rap-starter.service
-rw-r--r--  1 root root   56 Dec 23 12:24 bbb_record_core.slice
-rw-r--r--  1 root root  523 Dec 23 12:37 bbb-webrtc-sfu.service
-rw-r--r--  1 root root  293 Jan  3 14:14 disable-transparent-huge-pages.service
-rw-r--r--  1 root root  448 Dec 23 12:34 etherpad.service
-rw-r--r--  1 root root  503 Dec 23 12:37 kurento-media-server.service
~~~

**Comment by antobinary** [comment]
Resolved in #14110 on 2.4, the fix will be merged into 'develop' for 2.5 shortly. Thanks for the fixes @schrd !!
